src/renderer/services/auto-updater.service.ts

```diff
@@ -35,11 +55,68 @@ export class AutoUpdaterService {
         });
 
         this.progressService.show(this.downloadProgress$, true);
-        
         return promise;
     }
 
     public quitAndInstall() {
         this.ipcService.sendLazy("install-update");
     }
+
+    public getLastAppVersion(): string {
+        return this.configurationService.get("last-app-version");
+    }
+
+    public setLastAppVersion(){
+        lastValueFrom(this.getAppVersion()).then(v => this.configurationService.set("last-app-version", v));
+    }
+
+    private async getChangelog(): Promise<Changelog> {
+        try {
+        if (this.cacheChangelog) {
+            return this.cacheChangelog;
+        }
+
+        const path = `https://raw.githubusercontent.com/Zagrios/bs-manager/feature/add-changelog-modal/178/assets/jsons/changelogs.json`
+        const response = await fetch(path);
+        if (!response.ok) {
+            return undefined;
+        }
+
+        const data = await response.json();
+        if (!data) {
+            return undefined;
+        }
+
+        this.cacheChangelog = data;
+        return data;
+    }
+    catch(error){
+        this.ipcService.sendLazy("log-error", {args: error});
+        return undefined;
+    }
+    }
+
+    private async getChangelogVersion(version:string): Promise<ChangelogVersion> {
+        const changelogs = await this.getChangelog();
+        if (!changelogs) {
+            throw new Error(`No changelogs found`);
```

Comment:
If you do the requested changes in the `getChangelog` function; you will not need this if 'cause it should throw before